lib/imgix/path.rb

```diff
@@ -154,12 +154,23 @@ def build_srcset_pairs(options:, params:)
       srcset[0..-3]
     end
 
-    def build_dpr_srcset(params)
+    def build_dpr_srcset(options:, params:)
       srcset = ''
+
+      disable_variable_qualities = options['disable_variable_qualities'.to_sym] || false
+      validate_variable_qualities!(disable_variable_qualities)
+
       target_ratios = [1,2,3,4,5]
+      quality = params['q'.to_sym]
 
-      for ratio in target_ratios do
+      for index in 0..target_ratios.length - 1 do
+        ratio = target_ratios[index]
         params['dpr'.to_sym] = ratio
+
+        unless disable_variable_qualities
+          params['q'.to_sym] = quality || DPR_QUALITY[index]
+        end
+
```

Comment:
Would this be a little cleaner if `DPR_QUALITY` was a map instead of an array, with the target ratios as the keys?

Reply:
It's definitely more self-documenting, but then `index` won't align with `DPR_QUALITY` and `target_ratios` since we access the hash by the key and not an index like an array. So it would look like this:

```rb
# imgix.rb
DPR_QUALITY = {
    1 => 75, 
    2 => 50,
    3 => 35,
    4 => 23,
    5 => 20
  }

# path.rb
params['q'.to_sym] = quality || DPR_QUALITY[index + 1]
```

Reply:
Or you ditch the `index` thing entirely and go back to `for ratio in target_ratios do` above. then you could do `params['q'.to_sym] = quality || DPR_QUALITY[ratio]`, right?

Reply:
You're absolutely right 😂